scripts/G2-example.jl

```diff
@@ -0,0 +1,115 @@
+# # Example 3: Calculating observables
+
+# This is an example calculation of the two-body correlation function G_2.
+
+# A runnable script for this example is located 
+# [here](https://github.com/joachimbrand/Rimu.jl/blob/develop/scripts/G2-example.jl).
+# Run it with `julia G2-example.jl`.
+
+# Firstly, we load all needed modules.
+# `Rimu` for FCIQMC calculation, and `DataFrames` for output
+
+using Rimu
+using DataFrames
+
+# We use the same Hamiltonian as the first example,
+# a Bose-Hubbard model with 6 particles in 6 sites, with
+# strong interactions (we expect a Mott insulating state).
+m = n = 6
+aIni = near_uniform(BoseFS{n,m})
+H = HubbardReal1D(aIni; u = 6.0, t = 1.0)
+
+# Now we define the operators for the observables we wish to calculate
+dvals = 0:m-1
+G2list = ([G2RealCorrelator(d) for d in dvals]...,)
+# This is a tuple of [`G2RealCorrelator`](@ref)s, which are subtyped to 
+# [`AbstractHamiltonian`](@ref), but with less functionality than a full Hamiltonian.
+# It calculates the two-body correlation function on a lattice
+# ```math
+#     \hat{G}^{(2)}(d) = \frac{1}{M} \sum_i^M \hat{n}_i (\hat{n}_{i+d} - \delta_{0d}).
+# ```
+# with normalisation
+# ```math
+#     \sum_{d=0}^{M-1} \langle \hat{G}^{(2)}(d) \rangle = \frac{N (N-1)}{M}.
+# ```
+
+# Observables are calculated using the "replica trick" whereby several 
+# copies or "replicas" of the model are run simultaneously. We enable this by defining 
+# a [`ReplicaStrategy`](@ref). Each replica has its own state and FCIQMC is effectively 
+# performed independently on each one.
+# For calculating observables, we use [`AllOverlaps`](@ref) for the `ReplicaStrategy`. 
+# At each timestep, after the necessary FCIQMC variables are calculated for each replica,
+# (e.g. shift, norm etc.), this strategy calculates the overlaps of every operator with 
+# the wavefunctions from each pair of replicas. 
+num_reps = 3
+replica = AllOverlaps(num_reps; operator = G2list)
+
+# We need a reasonable number of timesteps to get good statistics, and we are running
+# multiple replicas, so we will only use a small number of walkers:
+steps_equilibrate = 1_000
+steps_measure = 5_000
+targetwalkers = 100;
+
+# Other FCIQMC parameters and strategies are the same as before
+dτ = 0.001
+reporting_interval = 1
+svec = DVec(aIni => 1)
```

Comment:
Hmm, this will select integer walker FCIQMC. Maybe we can leave it for now, but eventually we should make `IsDynamicSemistochastic` the default.

Reply:
I'll leave it for now so that it is the same as the first example. And the default can be changed behind the scenes rather thatn dealing with it in this example. This does suggest that there should be an example script that explains some of the different FCIQMC algorithm strategies.